.github/workflows/publish-docker.yml

```diff
@@ -0,0 +1,35 @@
+name: Docker Publish
+
+# Controls when the action will run. 
+on:
+  # Triggers the workflow on new SemVer tags
+  push:
+    tags: 
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+      - 'v[0-9]+.[0-9]+.[0-9]+-rc[0-9]+'
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: docker/setup-qemu-action@v1
+      - uses: docker/setup-buildx-action@v1
+      - uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - uses: crazy-max/ghaction-docker-meta@v2
```

Comment:
I still don't think this should be in the image build pipeline and instead should be in a separate (dependent) step, only passing the output.

Reply:
To me it seems to unnecessarily complicate the pipeline with no benefit.

Reply:
If Mr Max decides one day in the future that he wants to modify the action to swap the `/bin/docker` binary with one that contains malicious code to generate altered images, that would be a problem. I'd suggest pinning the git hash, but that is even more annoying than splitting out the job. The issue can be avoided by moving it out of the (image) build path and into its own job, passing only the output. There is no shared machine/environment state in that case, whereas there is shared state currently.

Reply:
That's one of the reasons I was suggesting building custom tooling in the first place. If that's the concern, I would much rather pin it to a specific SHA. Adding a dependent job doesn't completely block malicious intent.